finding/finding.go

```diff
@@ -52,45 +53,120 @@ type Location struct {
 type Outcome int
 
 const (
+	// NOTE: The additional '-' are intended for future use.
+	// This allows adding outcomes without breaking the values
+	// of existing outcomes.
 	// OutcomeNegative indicates a negative outcome.
 	OutcomeNegative Outcome = iota
+	_
```

Comment:
note: I have added some empty value to ensure we have a possibility to add outcomes, without changing existing values for backward compatibility.
These values are ordered from worst (OutcomeNegative =0) to best OutcomeNotSupported, and we can use comparison operators.
The ordering between OutcomeNotAvailable, OutcomeError, etc is a bit handwavy / subjective at this point. The ability to compare is useful for "structured" results (not this PR).
We may want to re-visit this before GA.

Reply:
It may also be worth explicitly setting the values of each outcome explicitly instead of using `iota`.

We could let us have gaps still, if we wanted to pursue having everything be ordered.

Reply:
added to https://github.com/ossf/scorecard/issues/2928.